docs/usage/bazel.md

```diff
@@ -5,22 +5,171 @@ description: Bazel dependencies support in Renovate
+- uses any files that are transitively imported by a `.bazelrc` file
+- only uses `--registry` flag values that are not associated with [a configuration](https://bazel.build/run/bazelrc#config)
+- queries the registries in the order that they are found in the `.bazelrc` file
+
+#### Example: multiple `.bazelrc` files
+
+In this example, there is a `.bazelrc` file which imports another file called `.registry.bazelrc`.
+Both files have `--registry` values:
+
+```
+# -------------
+# .bazelrc File
+# -------------
+import .registry.bazelrc
+build --registry=https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main
+
+# ----------------------
+# .registry.bazelrc File
+# ----------------------
+build --registry=https://example.com/custom_registry
+```
+
+The final registry list is:
+
+1. `<https://example.com/custom_registry>`
+1. `<https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main>`
+
+#### Example: registry entries using Bazel configuration
+
+In this example, a `.bazelrc` file has registry values with and without a configuration:
+
+```
+# -------------
+# .bazelrc File
+# -------------
+build:ci --registry=https://internal.server/custom_registry
+build --registry=https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main
+```
+
+In this case the `https://internal.server/custom_registry` is ignored.
+The final registry list is:
+
+1. `<https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main>`
+
+### Supported Bazel module declarations
+
+#### `bazel_dep`
+
+Renovate updates the `version` value for a [`bazel_dep`](https://bazel.build/rules/lib/globals/module#bazel_dep) declaration.
+
+```python
+bazel_dep(name = "cgrindel_bazel_starlib", version = "0.15.0")
+```
+
+In the example above, Renovate evaluates the `0.15.0` version against the repository's registries.
+If Renovate finds a newer version, it updates `0.15.0` to match that version.
+
+#### `git_override`
+
+If Renovate finds a [`git_override`](https://bazel.build/rules/lib/globals/module#git_override), it ignores the related `bazel_dep` entry and instead evaluates the `commit` value at the specified `remote`.
+
+```python
+bazel_dep(name = "cgrindel_bazel_starlib", version = "0.15.0")
+
+git_override(
+    module_name = "cgrindel_bazel_starlib",
+    commit = "fb47f0e9f7c376a7700fc9fe3319231ae57880df",
+    remote = "https://github.com/cgrindel/bazel-starlib.git",
+)
+```
+
+If the primary branch has a newer commit than in the list, Renovate updates the `commit` value.
+
+#### `single_version_override`
+
+The [`single_version_override`](https://bazel.build/rules/lib/globals/module#single_version_override) is a declaration with many purposes.
+Renovate only evaluates _two_ attributes from this declaration: `version` and `registry`.
+
+If a `version` is specified, it overrides the version specified in the `bazel_dep` pinning it to the specified value.
```

Comment:
This is a hard sentence to read, and also for me to rewrite. We're using the word `specified` three times.

Open for suggestions. 😉

Reply:
I trimmed it down. Does that work?